packages/orm/src/relations/CustomCollection.ts

```diff
@@ -14,14 +15,22 @@ export type CustomCollectionOpts<T extends Entity, U extends Entity> = {
   add?: (entity: T, other: U) => void;
   remove?: (entity: T, other: U) => void;
   /** Whether the reference is loaded, even w/o an explicit `.load` call, i.e. for DeepNew test instances. */
-  isLoaded: () => boolean;
+  isLoaded: (entity: T) => boolean;
```

Comment:
The pattern of `isLoaded = () => this...` meant the `isLoaded` lambda's `this` variable was bound to the fake instance on boot.

The old transform-properties technically rewrote the `isLoaded` lambda into the `getter`, so the `this` instance was still "the right entity".

But now that we do eval the lambda only once, and then reuse that `isLoaded` lambda across many instances (each with their own `relationConstructor.create` instance), we need the lambda to really not use `this`.